### PR TITLE
feat(cli): support recursive deletes

### PR DESCRIPTION
**Pull request by hsheth2**
Currently only works on single urns and requires browsepathv2 to be ingested for it to work correctly.

Closes  https://github.com/datahub-project/datahub/pull/8709.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: docs/how/delete-metadata.md [hunk 1]
```diff
@@ -43,6 +43,9 @@ datahub delete --platform snowflake
 # Filters can be combined, which will select entities that match all filters.
 datahub delete --platform looker --entity-type chart
 datahub delete --platform bigquery --env PROD
+
+# You can also do recursive deletes for some entity types.
```
Comment (asikowitz) [inline docs/how/delete-metadata.md:47]: Can we just specify those entities types, i.e. container and dataPlatformInstance

File: metadata-ingestion/src/datahub/ingestion/graph/client.py [hunk 2]
```diff
@@ -582,6 +599,36 @@ def get_urns_by_filter(
...
+                {
+                    "field": "platformInstance",
+                    "values": [platform_instance],
+                    "condition": "EQUAL",
+                }
+            ]
+
+        # Browse path v2 filter.
+        if container:
+            # TODO: Warn if container is not a fully qualified urn.
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/graph/client.py:622]: These warnings would be nice... and I'd probably just fail not even warn

**Review by asikowitz — APPROVED** [review]
Couple nits but LGTM

File: metadata-ingestion/src/datahub/ingestion/graph/client.py [hunk 3]
```diff
@@ -582,6 +599,36 @@ def get_urns_by_filter(
...
+
+            # TODO: Warn if platform_instance is not a fully qualified urn.
+
+            andFilters += [
+                {
+                    "field": "platformInstance",
+                    "values": [platform_instance],
+                    "condition": "EQUAL",
+                }
+            ]
```
[the remark this answers is not in the document]
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/graph/client.py:618]: the latter

File: metadata-ingestion/src/datahub/ingestion/graph/client.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/graph/client.py]: added

File: metadata-ingestion/src/datahub/ingestion/graph/client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (asikowitz) [reply metadata-ingestion/src/datahub/ingestion/graph/client.py:623]: We should properly produce this aspect then, like in a work unit processor

[the remark this answers is not in the document]
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/graph/client.py:623]: strongly agree - we're not consistent with it right now

**Pull request merged**